File: FlankSpeedLegacyDemo.Web/Models/Attachment.cs
namespace FlankSpeedLegacyDemo.Web.Models;

// Stub/metadata only — no actual file storage
public class Attachment
{
    public int Id { get; set; }
    public int WorkItemId { get; set; }
    public WorkItem? WorkItem { get; set; }
    public string FileName { get; set; } = string.Empty;
    public long FileSize { get; set; }
    public int UploadedByUserId { get; set; }
    public User? UploadedByUser { get; set; }
    public DateTime UploadedAt { get; set; } = DateTime.UtcNow;
}

File: power-apps-migration/src/models/Attachment.ts
/**
 * Migrated from: FlankSpeedLegacyDemo.Web/Models/Attachment.cs
 * Source: File attachment metadata (stub/metadata only in legacy — no actual file storage)
 * Target: Azure SQL [dbo].[Attachments] + SharePoint Document Library for actual files
 *
 * Migration note: In the Power Apps target, file storage should leverage
 * SharePoint document libraries via the SharePoint connector, with metadata
 * stored in Azure SQL for fast querying.
 */
import { User } from "./User";

export interface Attachment {
  id: number;
  workItemId: number;
  fileName: string;
  fileSize: number;
  uploadedByUserId: number;
  uploadedByUser?: User;
  uploadedAt: string;
}
